src/regexp.c

```diff
@@ -342,8 +342,14 @@ yp_regexp_parse_group(yp_regexp_parser_t *parser) {
     yp_regexp_options_init(&options);
 
     switch (*parser->cursor) {
-      case '#': // inline comments
-        return yp_regexp_char_find(parser, ')');
+      case '#': { // inline comments
+	bool found = yp_regexp_char_find(parser, ')');
+	// the close paren we found is escaped, we need to find another
+	while (parser->start <= parser->cursor - 2 && *(parser->cursor - 2) == '\\') {
```

Comment:
The RHS expression isn't considering encodings that `\\` can be a trailing byte, e.g. Shift_JIS.